pkg/generate/testdocs.go

```diff
@@ -107,6 +81,113 @@ func TestDocs(docPath string, pathToCheck string) error {
 	return err
 }
 
+// TestDoc is the documentation for a test case
+type TestDoc struct {
+	// name is the name of the test case
+	name string
+	// path is the relative path of the test file
+	path string
+	// lineStart is the starting line number of the testing function
+	lineStart int
+	// lineEnd is the ending line number of the testing function
+	lineEnd int
+	// isSubTest is true if the test case is a top-level test case, false if it's a validation method
+	isSubTest bool
+	// description is parsed from the function comment
+	description string
+	// steps are parsed from comments starting with `docs: `
+	steps []string
+	// specialCases are parsed from comments starting with `docs(special): `
+	specialCases []string
+	// specialCases are parsed from comments starting with `docs(skip): `
+	skips []string
+}
+
+// toMarkdown converts the TestDoc into a string in Markdown format
+func (d *TestDoc) toMarkdown() string {
+	b := &strings.Builder{}
+	if d.isSubTest {
+		b.WriteString("#### " + d.name + "\n")
+	} else {
+		b.WriteString("## " + d.name + "\n")
+	}
+
+	srcLinkText := fmt.Sprintf("%s#L%d-L%d", d.path, d.lineStart, d.lineEnd)
+	srcLinkURL := fmt.Sprintf("https://github.com/kubernetes/minikube/blob/master/%s#L%d-L%d", d.path, d.lineStart, d.lineEnd)
+	b.WriteString(fmt.Sprintf("Source code: [%s](%s)\n\n", srcLinkText, srcLinkURL))
```

Comment:
![Screenshot from 2021-06-16 00-24-52](https://user-images.githubusercontent.com/1311594/122157775-710bf980-ce39-11eb-9991-3d4d1c879a30.png)

Added a `source code` link for each test since that might be useful? Note that the URL links to the master branch git tree. When building the site from this branch the line numbers may not match the current master branch line numbers since this branch modifies the test file. However, when building the site from master branch the line numbers should match.

Reply:
Actually, I removed this `source code` for now since it requires developers to rerun `make generate-docs` every time they touch the testing files (because the line numbers will change, most likely) otherwise unit testing will fail. This can be a little bit annoying...